src/ActiveBackground.tsx

```diff
@@ -1,18 +1,34 @@
-import * as React from 'react'
+import type { ReactNode } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 import { useResizeObserver } from './hooks/useResizeObserver'
-import { IActiveBackground, IActiveBackgroundPattern } from './index'
 
-export const ActiveBackground: React.FC<IActiveBackground> = ({
-  pattern,
-  patternOptions = {},
-  className,
-  children,
-}) => {
-  const [canvasRef, setRef] = React.useState<HTMLCanvasElement | null>(null)
-  const onRefSet = React.useCallback((ref) => setRef(ref), [setRef])
-  const [update, setUpdate] = React.useState({})
+export type PatternOptions = Record<string, unknown>
```

Comment:
The types are all used here, so define them here.

`unknown` is safer than `any`.